Mage.Sets/src/mage/cards/s/SunfrillImitator.java

```diff
@@ -0,0 +1,120 @@
+package mage.cards.s;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.constants.*;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCreaturePermanent;
+import mage.util.functions.CopyApplier;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class SunfrillImitator extends CardImpl {
+
+    public SunfrillImitator(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{G}");
+        
+        this.subtype.add(SubType.DINOSAUR);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Whenever Sunfrill Imitator attacks, you may have it become a copy of another target Dinosaur you control, 
+        // except its name is Sunfrill Imitator and it has this ability.
+        this.addAbility(new SunfrillImitatorAbility());
+    }
+
+    private SunfrillImitator(final SunfrillImitator card) {
+        super(card);
+    }
+
+    @Override
+    public SunfrillImitator copy() {
+        return new SunfrillImitator(this);
+    }
+}
+
+class SunfrillImitatorAbility extends AttacksTriggeredAbility {
+
+    public SunfrillImitatorAbility() {
+        super(
+            new SunfrillImitatorEffect(),
+            true,
+            "Whenever Sunfrill Imitator attacks, you may have it become a copy of another target Dinosaur " +
+                    "you control, except its name is Sunfrill Imitator and it has this ability."
+        );
+        FilterCreaturePermanent dino_filter = new FilterCreaturePermanent(SubType.DINOSAUR, "dinosaur");
```

Comment:
1. Needs to be a FilterControlledPermanent, not a FilterCreaturePermanent (note that the rules text is "another target Dinosaur you control" which does not specify it being a creature)
2. filters should be declared static in the class, not within constructors or methods (unless they depend on other info which is not the same here)